src/tracing/esp32_trace/include/matter/tracing/macros_impl.h

```diff
@@ -21,17 +21,100 @@
 #error "Tracing macros seem to be double defined"
 #endif
 
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/core/CHIPError.h>
+
 namespace Insights {
 class ESP32Backend
```

Comment:
This class needs some documentation about purpose and what blacklist strings and hashes do and their difference.

Reply:
Also usage seems to be to place this inside a scope, so `backend` is not appropriate. One would assume a backend is something from https://github.com/project-chip/connectedhomeip/blob/master/src/tracing/backend.h

This looks like some form of `scope`. Why is this used instead of implementing an actual backend?

The overall implementation of the esp32_trace seems to need some updates to make it more inline with backend and macros logic in the rest of the systems.